Batteries/Control/OptionT.lean

```diff
@@ -0,0 +1,110 @@
+/-
+Copyright (c) 2017 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sebastian Ullrich
+-/
+
+import Batteries.Control.Lemmas
+
+/-!
+# Lemmas About Option Monad Transformer
+
+This file contains lemmas about the behavior of `OptionT` and `OptionT.run`.
+-/
+
+universe u v
+
+namespace Option
+
+variable {α β γ : Type u} {m : Type u → Type v} [Monad m]
+
+@[simp]
+theorem elimM_pure [LawfulMonad m] (x : Option α) (y : m β) (z : α → m β) :
+    Option.elimM (pure x) y z = Option.elim x y z := by
+  simp [Option.elimM, Option.elim]
+
+@[simp]
+theorem elimM_bind [LawfulMonad m] (x : m α) (f : α → m (Option β)) (y : m γ) (z : β → m γ) :
+    Option.elimM (x >>= f) y z = (do Option.elimM (f (← x)) y z) := by simp [Option.elimM]
+
+@[simp]
+theorem elimM_map [LawfulMonad m] (x : m α) (f : α → Option β) (y : m γ) (z : β → m γ) :
+    Option.elimM (f <$> x) y z = (do Option.elim (f (← x)) y z) := by simp [Option.elimM]
+
+end Option
+
+namespace OptionT
+
+variable {α β : Type u} {m : Type u → Type v} (x : OptionT m α)
+
+@[ext] theorem ext {x x' : OptionT m α} (h : x.run = x'.run) : x = x' := h
+
+@[simp] theorem run_mk (x : m (Option α)) : OptionT.run (OptionT.mk x) = x := rfl
+
+section Monad
+
+variable [Monad m]
+
+@[simp] theorem run_pure (a) : (pure a : OptionT m α).run = pure (some a) := rfl
+
+@[simp]
+theorem run_bind (f : α → OptionT m β) :
+    (x >>= f).run = Option.elimM x.run (pure none) (run ∘ f) := by
```

Comment:
Can you mention in the description that these were changed to use `elimM` as part of upstreaming from mathlib?